test/cmdlineargs.jl

```diff
@@ -828,3 +828,26 @@ end
         @test lines[4] == "bar"
     end
 end
+
+@testset "--heap-size-hint" begin
+    exename = `$(Base.julia_cmd())`
+    @test errors_not_signals(`$exename --heap-size-hint -e "exit(0)"`)
+    @testset "--heap-size-hint=$str" for str in ["asdf","","0","1.2vb","b","GB","1.2gb2","42gigabytes","5gig","2GiB","NaNt"]
+        @test errors_not_signals(`$exename --heap-size-hint=$str -e "exit(0)"`)
+        if !errors_not_signals(`$exename --heap-size-hint=$str -e "exit(0)"`)
+            @show str
+            try
+                run(`$exename --heap-size-hint=$str -e "exit(0)"`)
+            catch
+            end
+        end
```

Comment:
IIUC, this is what you were going for?
```suggestion
        heap_size_success = errors_not_signals(`$exename --heap-size-hint=$str -e "exit(0)"`)
        if !heap_size_success
            @show str
            run(ignorestatus(`$exename --heap-size-hint=$str -e "exit(0)"`))
        end
        @test heap_size_success
```
But I don't think we need the extra debugging code on CI

Reply:
bump